cypress/integration/mocked-okta/registerController.1.cy.ts

```diff
@@ -0,0 +1,131 @@
+      });
+      switch (status) {
+        case false:
+          specify(
+            "Then I should be shown the 'Check your email inbox' page",
+            () => {
+              cy.fixture('okta-responses/success/user').then(
+                (userResponse: ResponseFixture) => {
+                  // Set the correct user status on the response
+                  const response = { ...userResponse.response, status };
+                  cy.mockNext(userResponse.code, response);
+                  cy.get('button[type=submit]').click();
+                  cy.contains('Check your email inbox');
+                  cy.contains('Resend email');
+                },
+              );
+            },
+          );
+          break;
+        case 'ACTIVE':
+          specify(
+            "Then I should be shown the 'Check your email inbox' page",
+            () => {
+              cy.fixture('okta-responses/error/user-exists').then(
+                (userExistsError: ResponseFixture) => {
+                  cy.fixture('okta-responses/success/user').then(
+                    (userResponse: ResponseFixture) => {
+                      // Set the correct user status on the response
+                      const response = { ...userResponse.response, status };
+                      cy.mockNext(
+                        userExistsError.code,
+                        userExistsError.response,
+                      );
+                      cy.mockNext(userResponse.code, response);
+                      cy.get('button[type=submit]').click();
+                      cy.contains('Check your email inbox');
+                      cy.contains('Resend email');
+                    },
+                  );
+                },
+              );
+            },
+          );
+          break;
+        case 'PROVISIONED':
+        case 'STAGED':
+          // Then Gateway should generate an activation token
+          specify(
+            "Then I should be shown the 'Check your email inbox' page",
+            () => {
+              cy.fixture('okta-responses/error/user-exists').then(
+                (userExistsError: ResponseFixture) => {
+                  cy.fixture('okta-responses/success/user').then(
+                    (userResponse: ResponseFixture) => {
+                      cy.fixture('okta-responses/success/token').then(
+                        (tokenResponse: ResponseFixture) => {
+                          // Set the correct user status on the response
+                          const response = { ...userResponse.response, status };
+                          cy.mockNext(
+                            userExistsError.code,
+                            userExistsError.response,
+                          );
+                          cy.mockNext(userResponse.code, response);
+                          cy.mockNext(
+                            tokenResponse.code,
+                            tokenResponse.response,
+                          );
```

Comment:
Maybe it's because I'm not very familiar with the project but this type of mocking makes it kind of hard for me to follow the test.

As far as I understand we are just mocking all http requests and relying on the order of the requests to provide the correct mocked response, which is also pretty brittle as any change in the sequence of calls breaks the test and in some cases it can be hard to understand what each mocked response was originally meant for.

Could we use `mock_pattern` like [this](https://github.com/guardian/gateway/blob/main/cypress/integration/mocked/okta_register.3.cy.ts#L18-L36) instead ? It looks like you can provide at least the url you are mocking ( but not the http method as far as I can see) which makes more clear what is being mocked and it would make the test more robust.

If we cannot do that maybe a comment before each `mockNext` could make it easier to follow.

Of course there's a chance I misunderstood how all of this works and this whole comment doesn't make sense

Reply:
I totally agree - as we discussed in the call, I think it would be worth doing the following:

- Add comments to all the intercepts explaining why each mock is being used, what URL is being called, and what function is triggering that URL request.
- Consider using `mockPattern` instead to make it even more clear which URLs are being mocked, although I thought that it was worthwhile continuing to intercept single requests in order as it would make it very clear if a change modified the order of requests in the actual code.